lib/install/management/configure.go

```diff
@@ -110,13 +110,18 @@ func (d *Dispatcher) Configure(vch *vm.VirtualMachine, conf *config.VirtualConta
 		return err
 	}
 
-	if err = d.update(conf, settings, isConfigureOp); err == nil {
-		if conf.ShouldGrantPerms() {
-			err = GrantOpsUserPerms(d.ctx, d.session.Vim25(), conf)
-			if err != nil {
-				return errors.Errorf("Cannot init ops-user permissions, failure: %s. Exiting...", err)
-			}
+	err = d.update(conf, settings, isConfigureOp)
+
+	// If successful try to grant permissions to the ops-user
+	if err == nil && conf.ShouldGrantPerms() {
+		err = GrantOpsUserPerms(d.ctx, d.session.Vim25(), conf)
+		if err != nil {
+			// Fall through to roll back
+			err = errors.Errorf("Cannot init ops-user permissions, failure: %s", err)
 		}
+	}
+
+	if err == nil {
```

Comment:
There's some weird indenting going on here making it very unclear what the clean exit path is below. Could just be the PR rendering.